src/catalog/src/local/memory.rs

```diff
@@ -51,10 +51,10 @@ impl Default for MemoryCatalogManager {
         };
         let default_catalog = Arc::new(MemoryCatalogProvider::new());
         manager
-            .register_catalog("greptime".to_string(), default_catalog.clone())
+            .register_catalog_sync("greptime".to_string(), default_catalog.clone())
```

Comment:
I think it's also not necessary to add "sync" or "async" to methods' names in this file. The only reason I can think of is about performance. It might make not that big difference, while causes more complexity for callers. Let's just unify them to "async" version.

Reply:
`register_catalog_sync` and `register_schema_sync` is methods only applicable to `MemoryCatalogManager` and `MemoryCatalogProvider`, so that in test we don't have to call the async versions. Async is contagious.